runtime/jcl/common/thread.cpp

```diff
@@ -603,28 +603,110 @@ Java_java_lang_Thread_setExtentLocalCache(JNIEnv *env, jclass unusedClass, jobje
 #endif /* JAVA_SPEC_VERSION >= 19 */
 
 #if JAVA_SPEC_VERSION >= 20
+/**
+ * Frame walk function to find the most recent scoped value binding in the stack.
+ *
+ * @param currentThread the current thread
+ * @param walkState the stack walk state
+ * @return J9_STACKWALK_KEEP_ITERATING or J9_STACKWALK_STOP_ITERATING
+ */
+static UDATA
+findScopedValueBindingsWalkFunction(J9VMThread *currentThread, J9StackWalkState *walkState)
+{
+	UDATA rc = J9_STACKWALK_KEEP_ITERATING;
+	if (NULL != walkState->method) {
+		J9JavaVM *vm = currentThread->javaVM;
+		J9Method *method = walkState->method;
+		J9Class *methodClass = J9_CLASS_FROM_METHOD(method);
+		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+		J9UTF8 *methodName = J9ROMMETHOD_NAME(romMethod);
+
+		if (((J9VMJAVALANGTHREAD_OR_NULL(vm) == methodClass)
```

Comment:
Do all of the `runWith` have the same signature? If so, it might be better to do a name/sig text compare in order to avoid the OR_NULL issue of classes not being loaded.

Reply:
It also catches any new additions without needing to change the code.

Reply:
> Do all of the `runWith` have the same signature?

No, there are two overloads in ScopedValue class: `runWith(Snapshot newSnapshot, Callable<R> op)` and `runWith(Snapshot newSnapshot, Runnable op)`.

Reply:
Let me be more specific - do all the ones we're interested in have the same signature?

Reply:
Not quite sure what does "interested" mean here but I think those two overloads should be treated equally. Also, wouldn't the text comparison approach cause false positive (some other class have a runWith method with the same signature)?

Reply:
Yes, I suppose that's technically possible.